### PR TITLE
Bootstrapped Django 4.0.

### PR DESCRIPTION
**Pull request by felixxm**
(to be merged after the `stable/3.2.x` branch is created following Django 3.2 alpha)

### CODE REVIEW

File: tests/sessions_tests/tests.py [hunk 1]
```diff
@@ -346,6 +327,11 @@ def test_decode_failure_logged_to_security(self):
                 # The failed decode is logged.
                 self.assertIn('Session data corrupted', cm.output[0])
 
+    def test_decode_serializer_exception(self):
+        signer = TimestampSigner(salt=self.session.key_salt)
+        encoded = signer.sign(b'invalid data')
+        self.assertEqual(self.session.decode(encoded), {})
```
Comment (felixxm) [inline tests/sessions_tests/tests.py:333]: I added this test to cover catching exception from serializers in [decode()](https://github.com/django/django/commit/8d25fa4d80749ee74097108400bf63044a2a78d0#diff-3a6c11bbe36a0e6927f71ad8d669f0021897ba73768ee41073a318a12e11c3d1R103-R105).

File: tests/redirects_tests/tests.py [hunk 2]
```diff
@@ -58,12 +60,15 @@ def test_response_gone(self):
...
     def test_sites_not_installed(self):
+        def get_response(request):
+            return HttpResponse()
+
         msg = (
             'You cannot use RedirectFallbackMiddleware when '
             'django.contrib.sites is not installed.'
         )
         with self.assertRaisesMessage(ImproperlyConfigured, msg):
-            RedirectFallbackMiddleware()
```
Comment (felixxm) [inline tests/redirects_tests/tests.py:66]: It looks that we missed it in 4d973f593932285cd2f765400d915305d8e7333a.

**Review by carltongibson — COMMENTED** [review]
Good stuff. Thanks you @felixxm. 


Commit message for #15902 `s/don't/not/` but maybe: 

> Refs #15902 -- Stopped set_language() storing user's language in the session.

File: docs/releases/4.0.txt [hunk 3]
```diff
@@ -257,6 +257,9 @@ to remove usage of these features.
 * ``django.views.i18n.set_language()`` doesn't set the user language in
   ``request.session`` (key ``_language``).
 
+* ``alias=None`` is required in the signature of
+  ``django.db.models.Expression.get_group_by_cols()`` subclasses.
+
```
[the remark this answers is not in the document]
Comment (felixxm) [reply docs/releases/4.0.txt:262]: No, we added `alias=None` to the signature of `get_group_by_cols()` in 9dc367dc10594ad024c83d398a8e3c3f8f221446. Now it's required for all `Expression` subclasses

File: docs/releases/1.7.txt [hunk 4]
```diff
@@ -663,12 +663,11 @@ Internationalization
...
-  constant. Previously it was stored with the key ``django_language`` and the
-  ``LANGUAGE_SESSION_KEY`` constant did not exist, but keys reserved for Django
-  should start with an underscore. For backwards compatibility ``django_language``
-  is still read from in 1.7. Sessions will be migrated to the new key
-  as they are written.
+  accessed using the ``LANGUAGE_SESSION_KEY`` constant. Previously it was
+  stored with the key ``django_language`` and the ``LANGUAGE_SESSION_KEY``
+  constant did not exist, but keys reserved for Django should start with an
+  underscore. For backwards compatibility ``django_language`` is still read
+  from in 1.7. Sessions will be migrated to the new key as they are written.
```
[the remark this answers is not in the document]
Comment (felixxm) [reply docs/releases/1.7.txt:670]: It's release notes for Django 1.7, so ... but I'm open for suggestions :wink:

**Comment by felixxm** [comment]
> Commit message for #15902 s/don't/not/ but maybe:

Updated.

**Review by carltongibson — APPROVED** [review]
Looks good to go. Thanks for doing this @felixxm! 🥇

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Pull request merged**